packages/o/openssl/xmake.lua

```diff
@@ -83,6 +100,7 @@ package("openssl")
             table.insert(configs, "no-makedepend")
             table.insert(configs, "/FS")
         end
+        import("configure.patch")(package)
```

Comment:
Did you test all openssl versions? does this patch work for all versions?

Reply:
I tested the latest version (`1.1.1-w`) with the following combinations of targets and environments:

- Platform: Windows, Host: Windows
- Platform: MinGW, Host: MSYS2 (Git Bash)
- Platform: MinGW, Host: Linux
- Platform: Linux, Host: Linux

The patch does not work for versions `1.0.0` and `1.0.2-u`, resulting in the error:
```
error: @programdir\core\sandbox\modules\io.lua:188: cannot open file: Configurations/10-main.conf, Unknown Error (3)
```

Version `1.1.0-l` failed with the error:
```
Unsupported options: no-tests
```

The patch works for versions `1.1.1-h` and `1.1.1-p`.

Reply:
you can use `package:version():ge("1.1.1")` to limit patch

Reply:
OpenSSL versions prior to `1.1.1` do not support the `no-tests` build flag, and there is a conflict between `/FS` and `VC-WIN64A`. After removing these flags, I still could not build OpenSSL `1.1.0-l` for MinGW. While `1.1.0-l` builds for Windows, `1.0.2-u` does not.

Reply:
you can improve on_check to detect these cases.

Reply:
I discovered that OpenSSL `1.0` requires a different installation method. Detailed explanations can be found here: [How to Build OpenSSL 1.0 on Windows](https://github.com/pocoproject/poco/wiki/How-to-Build-OpenSSL-1.0-on-Windows).